pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -544,16 +544,25 @@ func (m *ManagerImpl) devicesToAllocate(podUID, contName, resource string, requi
 			return nil, fmt.Errorf("pod %q container %q changed request for resource %q from %d to %d", string(podUID), contName, resource, devices.Len(), required)
 		}
 	}
-	if needed == 0 {
-		// No change, no work.
-		return nil, nil
-	}
+
 	klog.V(3).InfoS("Need devices to allocate for pod", "deviceNumber", needed, "resourceName", resource, "podUID", string(podUID), "containerName", contName)
+	healthyDevices, hasRegistered := m.healthyDevices[resource]
+
 	// Check if resource registered with devicemanager
-	if _, ok := m.healthyDevices[resource]; !ok {
+	if !hasRegistered {
```

Comment:
Sure, we can do that and these are feasible options option but wanted to propose the least invasive change and go from there.

I believe, above and near `isDevicePluginResource` shouldn't be any problem but if we perform this check _in_  the `isDevicePluginResource`, we would be modifying the semantics and there are a few considerations:

1. If we return false for a device plugin resource that had previously registered and are not healthy, we are silently changing the semantics of `isDevicePluginResource`. The problem with this approach could be that  we are changing the semantics of this function and with the current implementation the pod would silently skip resource allocation rather than an explicit admission error (as it also includes cases corresponding to extended resources). 

1. If we modify `isDevicePluginResource`  to return an err and add the check there we should be okay. This way the case would manifest itself as an admission error but  we would have to perform this check at multiple places but we are expanding the scope from just allocation to hint generation logic as well which is not a problem but it is something worth noting.